CONTRIBUTING.md

```diff
@@ -135,6 +136,40 @@ def test_constructor_no_provider():
         LLMLoader(provider=None)
 ```
 
+### Checking what was printed and logged to stdout or stderr by the tested code
+
+It is possible to capture stdour and stderr by using standard fixture `capsys`:
+
+```python
+def test_foobar(capsys):
+    """Test the foobar function that prints to stdout."""
+    foobar("argument1", "argument2")
+
+    # check captured log output
+    captured_out = capsys.readouterr().out
+    assert captured_out == "Output printed by foobar function"
+    captured_err = capsys.readouterr().err
+    assert captured_err == ""
+```
+
+Capturing logs:
+
+```python
+@patch.dict(os.environ, {"LOG_LEVEL": "INFO"})
+def test_logger_show_message_flag(mock_load_dotenv, capsys):
+    """Test logger set with show_message flag."""
+    logger = Logger(logger_name="foo", log_level=logging.INFO, show_message=True)
+    logger.logger.debug("Debug message")
+
+    # check captured log output
+    # the log message should not be captured due to log level
+    # but the message should be shown
+    captured_out = capsys.readouterr().out
+    assert "Set LOG_LEVEL environment variable (e.g., INFO, DEBUG)" in captured_out
```

Comment:
Isn't it a bit weird, that you are not searching for the message that you logged at the beginning of the test?

Reply:
well it won' be there, I will update the example a bit

Reply:
I know it's a copied "real" test. But that test should rather test an empty output, right? Having an at least example clear would be great. :)

Reply:
fixed @onmete and TY for a good notice!